arches_controlled_lists/utils/skos.py

```diff
@@ -30,22 +30,36 @@ def __init__(self):
     def save_controlled_lists_from_skos(
         self,
         graph,
-        # overwrite_options="overwrite",
+        overwrite_options="overwrite",
     ):
         baseuuid = uuid.uuid4()
         allowed_languages = {}
         for lang in models.Language.objects.all():
             allowed_languages[lang.code] = lang
         default_lang = allowed_languages[settings.LANGUAGE_CODE]
 
+        existing_lists = List.objects.all()
+        existing_list_ids = [list.pk for list in existing_lists]
+        existing_list_items = ListItem.objects.all().prefetch_related(
+            "list_item_values"
+        )
+        existing_list_item_ids = [item.pk for item in existing_list_items]
+
         # if the graph is of the type rdflib.graph.Graph
         if isinstance(graph, Graph):
 
             ### Lists ###
             # Search for ConceptSchemes first - these will become Lists
             for scheme, v, o in graph.triples((None, RDF.type, SKOS.ConceptScheme)):
                 list_id = self.generate_uuidv5_from_subject(baseuuid, scheme)
-                new_list = List(list_id)
+
+                if list_id in existing_list_ids and overwrite_options == "ignore":
+                    new_list = List(uuid.uuid4())
+                elif list_id in existing_list_ids and overwrite_options == "overwrite":
+                    existing_lists.get(pk=list_id).delete()
```

Comment:
This changes the database state during the loop -- meaning if a delete happens, but on the second iteration the `list_id` is still in existing_lists, this get() will throw. Is this here to work around an issue?

Also, it's doing a direct db operation outside your transaction you have below wrapping most of the other writes.

Reply:
that seems right to me.: as i understand it, each skos:ConceptScheme should only have one node describing it